source/plugin/configuration/serialization.rst

```diff
@@ -135,6 +164,80 @@ The ``@ConfigSerializable`` annotation eliminates the need for any registration
 just generate an :javadoc:`ObjectMapper` for the class. The only limitation is that Configurate needs an empty
 constructor to instantiate a new object before filling in the annotated fields.
 
+.. note::
+
+    You can also have fields that are not are not annotated with ``@Setting`` in your ``@ConfigSerializable`` classes.
+    These fields won't be persisted to config files and can be used to store temporary references for your plugin.
+
+Using Default Values in ConfigSerializable Types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to use default values inside of ``@ConfigSerializable`` types, you just have to use Java's field
+initializers to set some default values; as long as the entry is not present in the config file the value won't be
+overwritten.
+
+.. code-block:: java
+
+    @ConfigSerializable
+    public class DiamondCounter {
+
+        @Setting(value="player", comment="Player UUID")
+        private UUID playerUUID;
+
+        @Setting(comment="Number of diamonds mined")
+        private int diamonds = 0;
+        
+        @Setting(comment="The time the player found a diamond last.")
+        private LocalDateTime diamonds = LocalDateTime.now();
+
+        [...]
+    }
+
+Example: Loading a ConfigSerializable Config with Default Values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of loading a default config from the plugin jar itself, it is also possible to just ask Configurate to create
+it if it is missing.
+
+.. code-block:: java
+
+    try {
+        this.config = this.configManager.load().<Configuration>getValue(Configuration.TYPE, Configuration::generateDefault);
+    } catch (ObjectMappingException | IOException e) {
+        this.logger.error("Failed to load the config - Using a default", e);
+        this.config = Configuration.generateErrorDefault();
+    }
+
+In this case you load the entire configuration into a ``Configuration`` object that contains all of your plugins
+configuration. Using such a class has the following benefits:
+
+* Type safety is guaranteed
+* No need to update the configuration file shipped in your plugin
+* You don't need to store lots of references for each of your configuration options
+* You can pass this config (or its parts) into methods or reference it from other classes
+* It is easy to write comments for each attribute in a place that also helps you during development
+
+.. note::
+
+    In this case ``Configuration.generateDefault()`` is called when the config file is missing or empty.
+    If you still want to load the shipped default config asset you can load it inside of that method.
+    ``Configuration.generateErrorDefault()`` is called when there was an error reading or parsing the config.
+    It is not necessary to use separate methods for those cases, you can also use the no-arg constructor in those cases,
```

Comment:
Turn the first comma into a semicolon. 
The second "in those cases" seems redundant.